exoscale/api/compute.py

```diff
@@ -0,0 +1,1714 @@
+            name (str): an instance template name to restrict results to
+            zone (Zone): a zone to restrict results to
+            type (str): an instance template type to restrict results to
+
+        Returns:
+            [InstanceTemplate]: list of instance templates
+        """
+
+        list = []
+
+        template_filters = {"exoscale": "featured", "mine": "self"}
+        if type not in template_filters.keys():
+            raise ValueError(
+                'invalid type "{}", supported types are: {}'.format(
+                    type, ", ".join(template_filters)
+                )
+            )
+
+        try:
+            resp = self.cs.listTemplates(
+                id=id,
+                name=name,
+                zoneid=getattr(zone, "id", None),
+                templatefilter=template_filters[type],
+            )
+            for i in resp.get("template", []):
+                list.append(InstanceTemplate.from_cs(self, i))
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"])
+
+        return list
+
+    ### Instance Type
+
+    def get_instance_types(self, name=None, id=None):
+        """
+        List available Compute instance types.
+
+        Parameters:
+            id (str): an instance type identifier to restrict results to
+            name (str): an instance type name to restrict results to
+
+        Returns:
+            [InstanceType]: list of instance types
+        """
+
+        list = []
+
+        try:
+            resp = self.cs.listServiceOfferings(id=id, name=name)
+            for i in resp.get("serviceoffering", []):
+                list.append(InstanceType.from_cs(i))
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"])
+
+        return list
+
+    ### Private Network
+
+    def create_private_network(
+        self, zone, name, description="", start_ip=None, end_ip=None, netmask=None
+    ):
+        """
+        Create a Private Network.
+
+        Parameters:
+            zone (Zone): the zone in which to create the Private Network
+            name (str): the Private Network name
+            description (str): the Private Network description
+            start_ip (str): the network IP range start address for managed
+                Private Networks
+            end_ip (str): the network IP range end address for managed
+                Private Networks
+            netmask (str): the network IP netmask for managed Private Networks
+
+        Returns:
+            PrivateNetwork: the Private Network created
+        """
+
+        try:
+            resp = self.cs.createNetwork(
+                zoneid=zone.id,
+                name=name,
+                displaytext=description,
+                startip=start_ip,
+                endip=end_ip,
+                netmask=netmask,
+                networkofferingid="eb35f4e6-0ecc-412e-9925-e469bf03d8fd",  # PrivNet
```

Comment:
Network offering ID is currently different for unmanaged vs managed privnets?

Reply:
Sorry no, but the requirement of passing a networkofferingid is about to disappear. The distinction will be purely based on the presence of startip/endip/netmask.

Reply:
OK I'll remove the hardcoded UUID once the change has been done in the API, because that the moment it's still required:

```
        try:
            resp = self.cs.createNetwork(
                zoneid=zone.id,
                name=name,
                displaytext=description,
                startip=start_ip,
                endip=end_ip,
                netmask=netmask,
            )
        except CloudStackApiException as e:
>           raise APIException(e.error["errortext"])
E           exoscale.api.APIException: Unable to execute API command createnetwork due to missing parameter networkofferingid

```